packages/chat-component/src/main.ts

```diff
@@ -187,6 +195,49 @@ export class ChatComponent extends LitElement {
     this.currentQuestion = this.questionInput.value;
   }
 
+  async handleCitationClick(sourceUrl: string, event: Event): Promise<void> {
+    if (sourceUrl?.endsWith('.md')) {
+      event?.preventDefault();
+
+      if (!this.isShowingThoughtProcess) {
+        this.selectedAsideTab = 'tab-citations';
+        this.showThoughtProcess();
+      }
+
+      const response = await fetch(sourceUrl);
+      if (response.ok) {
+        // highlight the clicked citation to make it clear which is being previewed
+        const citationsList = this.shadowRoot?.querySelectorAll(
+          '.aside .items__list.citations .items__listItem--citation',
+        );
+
+        if (citationsList) {
+          const citationsArray = [...citationsList];
+          const clickedIndex = citationsArray.findIndex((citation) => {
+            const link = citation.querySelector('a');
+            return link?.href === sourceUrl;
+          });
+
+          for (const citation of citationsList) {
+            const index = citationsArray.indexOf(citation);
+            if (index === clickedIndex) {
+              citation.classList.add('active');
+            } else {
+              citation.classList.remove('active');
+            }
+          }
+        }
+
+        // update the markdown previewer with the content of the clicked citation
+        const previewer = this.shadowRoot?.querySelector('#citation-previewer');
+        if (previewer) {
+          const markdownContent = await response.text();
+          previewer.innerHTML = DOMPurify.sanitize(md.render(markdownContent));
```

Comment:
this could also be done with reactive state, and injected with unsafeHtml()

Reply:
This seems to be hitting a few issues when updating not in the render function. For now, will continue using DOM.sanitize and will look at it separately.